src/lib.rs

```diff
@@ -45,3 +47,23 @@ mod log {
   pub(crate) use debug as trace;
   pub(crate) use debug as warn;
 }
+
+pub enum BuildId<'a> {
+  RawBytes(&'a [u8]),
+  Formatted(String),
```

Comment:
It seems a bit inconsistent to have one variant be borrowed and the other owned. My recommendation would be to use `Cow` in both, instead of forcing one or the other. The I think `formatted` could be allocation free in case the `BuildId::Formatted` variant is active.

Reply:
Makes sense, working on this one!

Reply:
Just pushed a separate commit for this one, I didn't add `Cow` to both types because we will always require an allocation to format the `RawBytes` variant, but perhaps I missed something. Never used cow before 😄 🐮 .  Let me know!

Reply:
Similar to your argument about a user already having a formatted build ID and not wanting to convert it back to "raw", they may or may not have a heap allocated one (owned). But if we only work with `String` and `Vec` we restrict ourselves to always allocate, even if conceptually we just need a temporary borrow internally. Similarly, if we only work with `&[u8]` or `&str` we may never be able to return a `BuildId` from any of our functions that could produce one down the line (I don't think we do that currently), because we wouldn't be able to allocate one and return it.

So using `&[u8]` and `&str` would be sufficient right now, but may become limiting in the future. Using `String` and `Vec` would mean always allocating, which seems suboptimal. `Cow` would provide a reasonable middle ground, I think, but I believe the argument applies to both variants. Does that make sense?

Reply:
> So using &[u8] and &str would be sufficient right now, but may become limiting in the future. 

Makes sense, thanks. I think this was the key point to understand where you are coming from. Will give another try at adding `Cow` to `RawBytes` on Monday.

Reply:
Sounds good. FYI, I enabled a few more lints and other minor changes in CI. So a few new warnings/errors may pop up once you rebase. Shouldn't be hard to fix, though.

Reply:
Thank you for the heads up! Just added some new commits that to address your feedback. Happy to fold them in 1 commit or feel free to squash them once this is ready 😄